packages/env-tests/src/shared/attestation.ts

```diff
@@ -0,0 +1,256 @@
+          known: true,
+        }
+      }
+
+      return
+    } catch (error) {
+      return { error, issuer: attestation.issuer, known: false }
+    }
+  })
+}
+
+export async function reportErrors(possibleErrors: RequestAttestationError[], logger: Logger) {
+  logger.info(
+    { possibleErrors: possibleErrors.filter((_) => _ && _.known).length },
+    'Reveal errors'
+  )
+
+  possibleErrors.filter(notEmpty).forEach((error) => {
+    if (error.known) {
+      logger.info({ ...error }, 'Error while requesting from attestation service')
+    } else {
+      logger.info({ ...error }, 'Unknown error while revealing to issuer')
+    }
+  })
+}
+
+// Inefficient, but should be fine for now
+// Could mark already checked messages in the future
+async function findValidCode(
+  attestations: AttestationsWrapper,
+  messages: string[],
+  identifier: string,
+  attestationsToComplete: ActionableAttestation[],
+  account: string
+) {
+  for (const message of messages) {
+    try {
+      const code = AttestationUtils.extractAttestationCodeFromMessage(message)
+      if (!code) {
+        continue
+      }
+
+      const issuer = await attestations.findMatchingIssuer(
+        identifier,
+        account,
+        code,
+        attestationsToComplete.map((_) => _.issuer)
+      )
+
+      if (!issuer) {
+        continue
+      }
+
+      const isValid = await attestations.validateAttestationCode(identifier, account, issuer, code)
+
+      if (!isValid) {
+        continue
+      }
+
+      return { code, issuer }
+    } catch {
+      continue
+    }
+  }
+
+  return
+}
+
+export async function getPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const phoneNumber = await chooseFromAvailablePhoneNumbers(twilioClient)
+
+  if (phoneNumber !== undefined) {
+    return phoneNumber
+  }
+
+  return createPhoneNumber(twilioClient, addressSid)
+}
+
+async function chooseFromAvailablePhoneNumbers(twilioClient: Twilio) {
+  const availableNumbers = await twilioClient.incomingPhoneNumbers.list()
+  const usableNumber = availableNumbers[0]
+  return usableNumber
+}
+async function createPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const countryCodes = ['GB', 'US']
+  const countryCode = sample(countryCodes)
+  const context = await twilioClient.availablePhoneNumbers.get(countryCode!)
+  const numbers = await context.mobile.list({ limit: 10 })
+  const usableNumber = numbers[0]
+
+  await twilioClient.incomingPhoneNumbers.create({
+    phoneNumber: usableNumber!.phoneNumber,
+    addressSid,
+    // Just an requestbin.com endpoint to avoid errors
+    smsUrl: 'https://enzyutth0wxme.x.pipedream.net/',
+  })
+
+  return usableNumber
+}
+
+async function fetchLatestMessagesFromToday(client: Twilio, phoneNumber: string, count: number) {
+  return client.messages.list({
+    to: phoneNumber,
+    pageSize: count,
+    // Twilio keeps track of dates in UTC so it could be yesterday too
+    dateSentAfter: moment()
+      .subtract(2, 'day')
```

Comment:
yesterday is 2 days ago?

Reply:
Copied from celotool again, but I believe yesterday is the day after the day two days ago :P